lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +469,53 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+
+    def get_chunk(self, trans, dataset, offset=0, ck_size=None):
+        bamfile = pysam.AlignmentFile(dataset.file_name, "rb")
```

Comment:
you probably want to use `with pysam.AlignmentFile(dataset.file_name, "rb") as bamfile:` here.